### PR TITLE
Implemented possibility to set custom Policy Combining Algorithm.

### PR DESCRIPTION
**Pull request by Yaroslav007**

### CODE REVIEW

File: entitlement-api/src/main/java/com/hevelian/identity/entitlement/api/PDPController.java [hunk 1]
```diff
@@ -63,11 +69,29 @@ public PDPPolicy getPolicy(@Valid @RequestBody PolicyIdDTO policyIdDTO)
...
+  }
+
+  // We use the allowableValues specially for Swagger UI. Since it requires the value to be a constant we cannot call
+  // a method from PolicyCombiningAlgorithmProvider bean to retrieve available Policy Combining Algorithm ids dynamically.
+  // Thus this value just needs to be in sync with the current provider instance.
+  @RequestMapping(path = "/setPolicyCombiningAlgorithm", method = RequestMethod.POST)
+  public PrimitiveResult<String> setPolicyCombiningAlgorithm(@ApiParam(value = "PDP policy combining algorithm", allowableValues = DenyOverridesPolicyAlg.algId + "," +
+      PermitOverridesPolicyAlg.algId + "," + FirstApplicablePolicyAlg.algId + "," + PermitUnlessDenyPolicyAlg.algId + "," + DenyUnlessPermitPolicyAlg.algId + "," +
+      OrderedPermitOverridesPolicyAlg.algId + "," + OrderedDenyOverridesPolicyAlg.algId + "," + OnlyOneApplicablePolicyAlg.algId + "," + "exception")
+      @RequestParam String algorithms) throws PDPPolicyCombiningAlgorithmNotSupportedException {
```
Comment (yuflyud) [inline entitlement-api/src/main/java/com/hevelian/identity/entitlement/api/PDPController.java]: algorithm

File: identity-server/src/main/java/com/hevelian/identity/server/config/rest/RestContextConfig.java [hunk 2]
```diff
@@ -85,6 +95,7 @@ public RestHandlerExceptionResolver restExceptionResolver() {
         // with wrong behavior.
         .contentNegotiationManager(contentNegotiationManager)
         .httpMessageConverters(requestMappingHandlerAdapter.getMessageConverters())
+        .messageInterpolator(messageInterpolator())
```
Comment (yuflyud) [inline identity-server/src/main/java/com/hevelian/identity/server/config/rest/RestContextConfig.java]: Add comment:
//Custom message interpolator to resolve Spring beans in messages.properties

File: entitlement-api/src/main/java/com/hevelian/identity/entitlement/api/PDPController.java (hunk 1 above)
Comment (Yaroslav007) [reply entitlement-api/src/main/java/com/hevelian/identity/entitlement/api/PDPController.java]: Fixed

File: identity-server/src/main/java/com/hevelian/identity/server/config/rest/RestContextConfig.java (hunk 2 above)
Comment (Yaroslav007) [reply identity-server/src/main/java/com/hevelian/identity/server/config/rest/RestContextConfig.java]: Fixed

**Pull request merged**